hr_job_categories/models/hr.py

```diff
@@ -0,0 +1,71 @@
+
+from odoo import models, fields, api
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class HRJob(models.Model):
+    _inherit = 'hr.job'
+
+    category_ids = fields.Many2many('hr.employee.category',
+                                    'job_category_rel',
+                                    'job_id',
+                                    'category_id',
+                                    string='Associated Tags')
+
+
+class HRContract(models.Model):
+    _inherit = 'hr.contract'
+
+    @api.multi
+    def _remove_tags(self, employee_id=None, job_id=None):
+        if not employee_id or not job_id:
+            return
+        employee = self.env['hr.employee'].browse(employee_id)
+        empl_tags = employee.category_ids
+        job = self.env['hr.job'].browse(job_id)
+        _logger.debug('Removing employee tags if tag exists on contract '
+                      'job: %s', empl_tags)
+        for tag in job.category_ids:
```

Comment:
Refactoring is out of the scope of the PR but pls add a TODO comment to refactor these methods to write only once.